tests/rptest/archival/s3_client.py

```diff
@@ -51,6 +51,10 @@ def __init__(self,
                  logger,
                  endpoint=None,
                  disable_ssl=True):
+
+        logger.debug(
+            f"Constructed S3Client in region {region}, endpoint {endpoint} using key = {access_key is not None}"
```

Comment:
What is `key = {access_key is not None}` expected to print?

Reply:
i think "key = True" in the case access_key is provided. probably to not leak the access key in the logs?

Reply:
I think it might be worth stating that in the log explicitly eg `access_key is set = ...`

Reply:
Andrea is right, but it was a bit ambiguous.  I like Abhijat's suggestion, used that instead.